### PR TITLE
fix(core): 9717 Get default layers for embedded map, use extent, fix KLA prefix

### PR DESCRIPTION
**Pull request by vkozio**
https://kontur.fibery.io/Tasks/Task/FE-Get-default-layers-for-embedded-map-9717
https://kontur.fibery.io/Tasks/Task/DN2-FE.-Remove-layers-ids-KLA-prefix-14466
https://kontur.fibery.io/Tasks/Task/Configure-optional-initial-map-extent-for-apps-14091

appConfig constructed from new endpoint
removed irrelevant logout tasks
currentLanguageAtom removed, not needed anymore

### CODE REVIEW

File: src/components/ConnectedMap/map-libre-adapter/index.tsx [hunk 1]
```diff
@@ -141,6 +142,10 @@ function MapboxMap(
       dispatchMetricsEvent(EVENT_MAP_IDLE, true);
     });
     setMap(mapInstance);
+    // use initial extent if no coords in url
+    if (!currentMapPosition && appConfig.extent) {
```
Comment (IlyaIzr) [inline src/components/ConnectedMap/map-libre-adapter/index.tsx:146]: i see `LAYERS_BY_DEFAULT` was removed from config but no `extent` was added. Should it be added or should it be on cd-repo config?

File: src/core/app/authHooks.ts [hunk 2]
```diff
@@ -1,58 +1,25 @@
...
+      const mergedUserdata = {
+        ...appConfig.user,
+        id: response.jwtData.sub,
+        token: response.token,
+      };
+      currentUserAtom.setUser.dispatch(mergedUserdata);
+    }
   }
+  setFeatures(appConfig.effectiveFeatures);
 }
```
Comment (IlyaIzr) [inline src/core/app/authHooks.ts]: 1. response is now optional, but there are no actions when it's not provided - weird
2. `setFeatures` called even when no response and config didn't change, is that expected?
3. if above thigs expected why not remove extra nest level
```suggestion
export async function onLogin(response?: AuthSuccessResponse) {
  if (response && appConfig.user) {
    externalLoginTasks(appConfig.user);
    const mergedUserdata = {
      ...appConfig.user,
      id: response.jwtData.sub,
      token: response.token,
    };
    currentUserAtom.setUser.dispatch(mergedUserdata);
  }
  setFeatures(appConfig.effectiveFeatures);
```

File: src/core/app/init.ts [hunk 3]
```diff
@@ -17,56 +19,59 @@ export async function appInit() {
...
-  // Stage - 1
-  // get application ID from URL or API
-  // load custom app config from api
-  const appId = initialState.app || (await apiClient.get<string>('/apps/default_id'));
-  if (!appId) {
-    // cannot continue without appId
-    throw new Error('Error getting application ID');
-  }
-  initialState.app = appId;
+  const localAuthToken = authClientInstance.checkLocalAuthToken();
```
Comment (IlyaIzr) [inline src/core/app/init.ts]: variable `localAuthToken` not used anywhere

File: src/core/app/init.ts [hunk 4]
```diff
@@ -17,56 +19,59 @@ export async function appInit() {
...
+    { appId: initialState.app },
+    true,
   );
 
   if (!appInfo) {
     // cannot continue without custom app config
     throw new Error('Error getting application config');
   }
 
+  initialState.app = appInfo.id;
```
Comment (IlyaIzr) [inline src/core/app/init.ts]:
```suggestion
  // BE returns default appId if it wasn't presented on request
  initialState.app = appInfo.id;
```

File: src/core/app/init.ts [hunk 5]
```diff
@@ -17,56 +19,59 @@ export async function appInit() {
...
-    throw new Error('Error getting application ID');
-  }
-  initialState.app = appId;
+  const localAuthToken = authClientInstance.checkLocalAuthToken();
 
-  const appInfo = await apiClient.get<AppInfoResponse>(
-    `/apps/${appId}`,
-    undefined,
-    false,
+  const appInfo = await apiClient.get<AppConfiguration>(
```
Comment (IlyaIzr) [inline src/core/app/init.ts]: naming suggestion:
we already have `appConfig` on this file, `appInfo` sounds very similar to it
how about smth like `remoteAppInfo` or ``remoteAppData`?

File: src/core/app/init.ts [hunk 6]
```diff
@@ -17,56 +19,59 @@ export async function appInit() {
...
     // cannot continue without custom app config
     throw new Error('Error getting application config');
   }
 
+  initialState.app = appInfo.id;
+  if (!appConfig.user) {
+    appConfig.user = defaultUserProfileData;
+  }
+
+  setAppLanguage(appConfig.user.language);
```
Comment (IlyaIzr) [inline src/core/app/init.ts]: why not move  
```
if (!appConfig.user) {
    appConfig.user = defaultUserProfileData;
  }
```
into `setAppLanguage`?

File: src/core/app/types.ts [hunk 7]
```diff
@@ -73,19 +73,28 @@ export interface AppConfigGlobal {
...
-  effectiveFeatures: Record<string, boolean>;
+  effectiveFeatures: Record<string, object | boolean>;
 }
 
-// from api
-export type AppInfoResponse = {
+export type AppConfigEffective = AppConfigGlobal & AppConfiguration;
+
+// Unified config from /app/configuration
+export type AppConfiguration = {
```
Comment (IlyaIzr) [inline src/core/app/types.ts:82]: Me again with naming
There's no way to differ `AppConfigGlobal` and  `AppConfiguration` if you don't know what they are
I suggest renaming AppConfiguration to RemoteAppConfig or something
We can also rename AppConfigGlobal to StaticAppConfig or smth

File: src/core/app/user.ts [hunk 8]
```diff
@@ -1,25 +1,33 @@
 import { appConfig } from '~core/app_config';
 
-export type CurrentUser = {
-  id: string | null;
+export type UserProfileMain = {
```
Comment (IlyaIzr) [inline src/core/app/user.ts:3]: `UserProfileMain` is toooo abstract
UserProfileCredentials / UserProfileCreds may be?

File: src/core/app/user.ts [hunk 9]
```diff
@@ -1,25 +1,33 @@
...
   language: string;
   useMetricUnits: boolean;
   subscribedToKonturUpdates: boolean;
-  bio?: string;
   osmEditor: string;
   defaultFeed: string;
   theme: string;
+};
+
+export type UserProfileApi = UserProfileMain & UserProfileSettings;
```
Comment (IlyaIzr) [inline src/core/app/user.ts:21]:
```suggestion
export type UserProfileApiResponse = UserProfileMain & UserProfileSettings;
```
?
Or just `UserProfileResponse`

File: src/core/auth/client/AuthClient.ts [hunk 10]
```diff
@@ -86,31 +79,36 @@ export class AuthClient {
...
+      return true;
+    }
+    // anon init flow
+    this.startPublic();
+    return false;
+  }
+
+  startPublic() {
+    over(this.loginHooks)();
+  }
```
Comment (IlyaIzr) [inline src/core/auth/client/AuthClient.ts]: i like anonimus word here very much!
may be method should be called `startAnonymosly` too?

**Comment by IlyaIzr** [comment]
General code LGTM, i've added lot of naming suggestion
However try to change laguage for user - it's not working for me
@vkozio please check that

**Comment by IlyaIzr** [comment]
Also before deploying config in cd-repo MUST be changed
"LAYERS_BY_DEFAULT" should be removed
I've also got error on local build for "INTERCOM_DEFAULT_NAME" being `null`

File: src/core/app/authHooks.ts (hunk 2 above)
Comment (vkozio) [reply src/core/app/authHooks.ts]: 1 - now it combines logic for auth and anon init
2 yes, to be refactored in other tasks
3 yes, thanks

File: src/core/auth/client/AuthClient.ts (hunk 10 above)
Comment (vkozio) [reply src/core/auth/client/AuthClient.ts]: why not)

File: src/core/app/user.ts (hunk 9 above)
Comment (vkozio) [reply src/core/app/user.ts:21]: to be refactored in next iterations

File: src/core/app/user.ts (hunk 8 above)
Comment (vkozio) [reply src/core/app/user.ts:3]: to be refactored in next iterations

File: src/core/app/types.ts (hunk 7 above)
Comment (vkozio) [reply src/core/app/types.ts:82]: to be refactored in next iterations

**Review by IlyaIzr — APPROVED** [review]

**Pull request merged**